projects/dsp-ui/src/lib/search/expert-search/expert-search.component.ts

```diff
@@ -0,0 +1,117 @@
+import { Component, EventEmitter, Inject, Input, OnInit, Output } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { KnoraApiConfig } from '@dasch-swiss/dsp-js';
+import { DspApiConfigToken } from '../../core/core.module';
+import { AdvancedSearchParams, AdvancedSearchParamsService } from '../services/advanced-search-params.service';
+
+@Component({
+    selector: 'dsp-expert-search',
+    templateUrl: './expert-search.component.html',
+    styleUrls: ['./expert-search.component.scss']
+})
+export class ExpertSearchComponent implements OnInit {
+
+    /**
+     * @param gravsearchQuery Send the gravsearch query back.
+     */
+    @Output() gravsearchQuery = new EventEmitter<string>();
+
+    /**
+     * @param toggleExpertSearchForm Trigger toggle for extended search form.
+     */
+    @Output() toggleExpertSearchForm = new EventEmitter<boolean>();
+
+    expertSearchForm: FormGroup;
+
+    constructor(
+        @Inject(DspApiConfigToken) private dspApiConfig: KnoraApiConfig,
+        private _searchParamsService: AdvancedSearchParamsService,
+        private fb: FormBuilder,
+        private _route: ActivatedRoute,
+        private _router: Router
+    ) { }
+
+    ngOnInit(): void {
+        this.initForm();
+    }
+
+    /**
+     * @ignore
+     * Initiate the form with predefined Gravsearch query as example.
+     */
+    private initForm() {
+        this.expertSearchForm = this.fb.group({
+            gravquery: [
+`PREFIX knora-api: <http://api.knora.org/ontology/knora-api/simple/v2#>
```

Comment:
please assign the query string to a variable and then use the variable here.

Reply:
done in ea9edea

Reply:
ok